### PR TITLE
More robust handling of large numbers of projects

### LINKED ISSUE #389: ProjectsSummary leaking watches
The projects watch here is never removed.

https://github.com/openshift/origin-web-catalog/blob/master/src/components/projects-summary/projects-summary.controller.ts#L106

@jeff-phillips-18 FYI, I can fix since I'm in the code anyway.

### PR DESCRIPTION
**Pull request by spadgett**
- Don't try to watch extremely large numbers of projects
- Use cached project list for select-project component
- Let users type a project name if the project list failed to load

See https://github.com/openshift/origin-web-console/issues/1923
Requires https://github.com/openshift/origin-web-common/pull/150
Updates origin-web-common to 0.0.47

Closes #389

### CODE REVIEW

File: src/components/projects-summary/projects-summary.controller.ts [hunk 1]
```diff
@@ -102,8 +112,26 @@ export class ProjectsSummaryController implements angular.IController {
     });
   }
 
+  public $onDestory() {
```
Comment (jeff-phillips-18) [inline src/components/projects-summary/projects-summary.controller.ts]: Typo here.

**Comment by jeff-phillips-18** [comment]
We will likely need to update the mock project service as well. Can be done in a separate PR.

File: src/components/projects-summary/projects-summary.controller.ts (hunk 1 above)
Comment (spadgett) [inline src/components/projects-summary/projects-summary.controller.ts]: Yikes, thanks for catching that

**Comment by spadgett** [comment]
Removing WIP. This should be ready now.

**Comment by dtaylor113** [comment]
Looks good @spadgett, can you add a unit test to verify HTML when $ctrl.isProjectListIncomplete?  
Thanks,
Dave

**Comment by spadgett** [comment]
Good comment @dtaylor113. I added a test. Thanks.

@jeff-phillips-18 Look OK if Travis passes?

**Comment by jeff-phillips-18** [comment]
@spadgett LGTM

**Pull request merged**